scalableminds/webknossos
# Improve Bounding Box Selection

### URL of deployed dev instance (used for testing):
- https://improvebboxselection.webknossos.xyz/

### Steps to test:
- Open an annotation with some bounding boxes
- Within a view port (except 3D), right-click on a bounding box edge. In the context menu, a new entry should appear, that lets you focus the BB in the BB tab (if open). Make sure this work. If the list entry wasnt visible due to scrolling, clicking this new context menu action should focus it
- Hovering above the edge of a bounding box should highlight it
- Within the move tool, left clicking the edge of a bounding box should focus it in the bounding box tab and make it the active bounding box
- This behaviour (on click) should not interfere with other tools. Although the highlighting of the edges is still active in other tools, similar to the highlighting of segments.

### TODOs:
- [x]  Allow to select a bbox in move tool via ctrl + click (just like in bbox tool)
    - [x] what about event.shiftKey || !useLegacyBindings
- [ ] ~~Allow to select a bbox in 3d viewport~~ is a separate issue
- [x] In any tool, the bbox borders are highlighted upon hovering. Why? Is this good? -> "If a tool does not define a specific mouse binding, the bindings of the MoveTool are used as a fallback."


### Issues:
- fixes #8004 

------
(Please delete unneeded items, merge only when none are left open)
